enhancements/monitoring/scrape-profiles.md

```diff
@@ -0,0 +1,407 @@
+spec:
+  serviceMonitorSelector:
+    matchExpressions:
+    - key: monitoring.openshift.io/scrape-profile
+      operator: NotIn
+      values:
+      - "full"
+```
+ 
+An OpenShift team that wants to support the scrape profiles feature would need
+to provide 2 monitors for each profile (in this example 1 service monitor per
+profile).
+
+```yaml
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    k8s-app: telemeter-client
+    monitoring.openshift.io/scrape-profile: full
+  name: telemeter-client
+  namespace: openshift-monitoring
+spec:
+  endpoints:
+  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+    interval: 30s
+    port: https
+    scheme: https
+    tlsConfig:
+      <...>
+  jobLabel: k8s-app
+  selector:
+    matchLabels:
+      k8s-app: telemeter-client
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    k8s-app: telemeter-client
+    monitoring.openshift.io/scrape-profile: minimal
+  name: telemeter-client
+  namespace: openshift-monitoring
+spec:
+  endpoints:
+  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+    interval: 30s
+    port: https
+    scheme: https
+    tlsConfig:
+      <...>
+  jobLabel: k8s-app
+  selector:
+    matchLabels:
+      k8s-app: telemeter-client
+  metricRelabelings:
+  - sourceLabels: [__name__]
+    action: keep
+    regex: "federate_samples|federate_filtered_samples"
+ ```
+
+Note: the `metricRelabeling` section only keeping two metrics, while the rest is
+dropped.
+
+Finally, a team that adopts the scrape profile feature should also add 
+themselves and their component to the list under 
+[Infrastruture needed](#infrastructure-needed-optional)
+
+#### Variation [optional]
+
+NA
+
+### API Extensions
+
+- Addition of a new field to the `cluster-monitoring-config` ConfigMap
+
+```go
+type PrometheusK8sConfig struct {
+  // Defines the scraping profile that will be enforced on the platform
+  // prometheus instance. Possible values are full and minimal.
+	ScrapeProfile string `json:"scrapeProfile,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+Each OpenShift team that wants to adopt this feature will be responsible for
+providing the different monitors. This work is not trivial. Dependencies between
+operators and their metrics exist. This makes it difficult for developers to
+determine whether a given metric must be added to the "minimal" profile or not.
+To aid teams with this problem we will provide a tool that would consume a
+monitor and generate the "minimal" monitor for that operator. However for this,
+the tool would need to have access to an up to date, installation of OpenShift
+in order to query the Prometheus instance running in the cluster.
+
+### Risks and Mitigations
+
+- How are monitors supposed to be kept up to date? In 4.12 a metric that wasn't
+  being used in an alert is now required, how does the monitor responsible for
```

Comment:
As discussed this will be done by a CI test that will validate that all resources (Alerts, PrometheusRule and Dashboards) are metrics that are present in the ServiceMonitors